harmoni_core/harmoni_pattern/test/test_sequential_speak.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+
+
+PKG = "test_harmoni_pattern"
+# Common Imports
+import unittest, rospy, roslib, sys
+
+# Specific Imports
+from actionlib_msgs.msg import GoalStatus
+from harmoni_common_lib.action_client import HarmoniActionClient
+from harmoni_common_lib.constants import (
+    DetectorNameSpace,
+    SensorNameSpace,
+    ActionType,
+    State,
+)
+from harmoni_common_msgs.msg import harmoniAction, harmoniFeedback, harmoniResult
+from audio_common_msgs.msg import AudioData
+from std_msgs.msg import String
+import time
+import os, io
+
+
+class TestSequential(unittest.TestCase):
+    """This test of the sequential pattern player uses a simple mic test script.
+
+    If the mic is launched and starts successfully (as verified by the subscriber)
+    then the test is passed.
+    """
+
+    def __init__(self, *args):
+        super(TestSequential, self).__init__(*args)
+        self.feedback = State.INIT
+        self.result = False
+        rospy.init_node("test_sequence", log_level=rospy.INFO)
+        self.rate = rospy.Rate(20)
+
+    def setUp(self):
+        self.server = "speak_test_default"
+        self.client = HarmoniActionClient(self.server)
+        self.client.setup_client(
+            self.server, self._result_callback, self._feedback_callback, wait=True
+        )
+        rospy.loginfo("TestSequential: Turning ON sequential server")
+        self.client.send_goal(
+            action_goal=ActionType.ON, optional_data="Setup", wait=False
+        )
+        rospy.loginfo("TestSequential: Started up. waiting for sequence startup")
+
+        # wait for start state
+        # while not rospy.is_shutdown() and self.feedback != State.START:
+        #     self.rate.sleep()
+        time.sleep(3)
+
+    def _feedback_callback(self, data):
+        rospy.loginfo(f"TestSequential: Feedback: {data}")
+        self.feedback = data["state"]
+
+    def _result_callback(self, data):
+        rospy.loginfo(f"TestSequential: Result: \n{data}")
+        self.result = True
+
+
+class TestSequential_Valid(TestSequential):
+    def test_IO(self):
+        rospy.loginfo(
+            "TestSequential[TEST]: basic test to ensure pattern can start a service "
+        )
+        while not rospy.is_shutdown() and not self.result:
+            self.rate.sleep()
+        assert self.result == True
```

Comment:
Normally you would want to separate out your test functions so that you know which one passes and which one does not...I suppose since it's not automated and requires user interaction though, you can just do a single test function and print out statements based on progress through the test.